os/include/tinyara/audio/audio.h

```diff
@@ -0,0 +1,727 @@
+#define AUDIO_FU_DELAY              0x0080
+#define AUDIO_FU_BASS_BOOST         0x0100
+#define AUDIO_FU_LOUDNESS           0x0200
+#define AUDIO_FU_INP_GAIN           0x0400
+#define AUDIO_FU_BALANCE            0x0800
+#define AUDIO_FU_PHASE_INVERT       0x1000
+#define AUDIO_FU_UNDERFLOW          0x2000
+#define AUDIO_FU_OVERFLOW           0x4000
+#define AUDIO_FU_LATENCY            0x8000
+
+/* Processing Unit controls *************************************************/
+
+#define AUDIO_PU_UNDEF              0x00
+#define AUDIO_PU_UPDOWNMIX          0x01
+#define AUDIO_PU_DOLBY_PROLOGIC     0x02
+#define AUDIO_PU_STEREO_EXTENDER    0x03
+#define AUDIO_PU_SUBSAMPLE_FORWARD  0x04
+#define AUDIO_PU_SUBSAMPLE_REWIND   0x05
+
+/* Stereo Extender PU Controls **********************************************/
+
+#define AUDIO_STEXT_UNDEF           0x00
+#define AUDIO_STEXT_ENABLE          0x01
+#define AUDIO_STEXT_WIDTH           0x02
+#define AUDIO_STEXT_UNDERFLOW       0x03
+#define AUDIO_STEXT_OVERFLOW        0x04
+#define AUDIO_STEXT_LATENCY         0x05
+
+/* Audio Callback Reasons ***************************************************/
+
+#define AUDIO_CALLBACK_UNDEF        0x00
+#define AUDIO_CALLBACK_DEQUEUE      0x01
+#define AUDIO_CALLBACK_IOERR        0x02
+#define AUDIO_CALLBACK_COMPLETE     0x03
+
+/* Audio Pipeline Buffer (AP Buffer) flags **********************************/
+
+#define AUDIO_ABP_ALIGNMENT         0x000f	/* Mask to define buffer alignment */
+#define AUDIO_ABP_CANDMA            0x0010	/* Set if the data is DMA'able */
+#define AUDIO_ABP_STATIC            0x0020	/* Set if statically allocated */
+#define AUDIO_ABP_ACTIVE            0x0040	/* Set if this buffer is still active.
+											 *   A buffer could become inactive
+											 *   if it is processed by an output
+											 *   device or a processing device
+											 *   that replaces it with an alternate
+											 *   buffer as a result of some DSP
+											 *   operation, etc.
+											 */
+
+/* Standard Audio Message Queue message IDs */
+
+#define AUDIO_MSG_NONE              0
+#define AUDIO_MSG_DEQUEUE           1
+#define AUDIO_MSG_START             2
+#define AUDIO_MSG_STOP              3
+#define AUDIO_MSG_PAUSE             4
+#define AUDIO_MSG_RESUME            5
+#define AUDIO_MSG_DATA_REQUEST      6
+#define AUDIO_MSG_ENQUEUE           7
+#define AUDIO_MSG_COMPLETE          8
+#define AUDIO_MSG_USER             64
+
+/* Audio Pipeline Buffer flags */
+
+#define AUDIO_APB_OUTPUT_ENQUEUED   (1 << 0)
+#define AUDIO_APB_OUTPUT_PROCESS    (1 << 1)
+#define AUDIO_APB_DEQUEUED          (1 << 2)
+#define AUDIO_APB_FINAL             (1 << 3)	/* Last buffer in the stream */
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* Define the size of AP Buffer sample count base on CONFIG */
+
+#ifdef CONFIG_AUDIO_LARGE_BUFFERS
```

Comment:
Unless audio framework layer is developed these configuration parameters wont appear in Kconfig. Then why is it needed now? There is a strong dependency with I2S device, we need this to develop I2S for now.